### PR TITLE
Add support for Postgresql

### PR DESCRIPTION
**Pull request by theundeadmonk**
🚧 

Would you guys be interested in this?

### CODE REVIEW

**Comment by jprobinson** [comment]
Howdy @theundeadmonk!

This actually looks like a nice addition to the `config` package. Go ahead and clean it up and I'll look to bring it in.

**Comment by theundeadmonk** [comment]
Awesome. I'll split this up into two PR's one with the postgres config and one with the combined config since it has a dependency on the postgres configs. Let me finish testing the changes on my machine and I should have the PR's soon.

**Comment by jprobinson** [comment]
You should be able to do it in a single PR. Just clone your fork into `$GOPATH/src/github.com/NYTimes/` and do your work there so the import paths work.

**Comment by theundeadmonk** [comment]
@jprobinson Tested it on my machine with Postgres and it worked fine. It's ready for a code review.

File: config/doc.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ The subpackages contain structs meant for managing common configuration options
 * MySQL
 * MongoDB
 * Oracle
+* Postgresql
```
Comment (jprobinson) [inline config/doc.go]: `PostgreSQL` here too, please.

File: config/combined/config.go [hunk 2]
```diff
@@ -34,6 +35,8 @@ type Config struct {
 
 	Oracle oracle.Config
 
+	Postgresql *postgresql.Config
```
Comment (jprobinson) [inline config/combined/config.go]: Just a nitpick; please change the variable name to be camel case: `PostgreSQL`

File: config/postgresql/postgresql.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import (
+	"database/sql"
+	"fmt"
+	"net/url"
+
+	"github.com/NYTimes/gizmo/config"
+)
+
+// Config holds everything yo need to
+// connect and interact with a Postgresql DB
```
Comment (jprobinson) [inline config/postgresql/postgresql.go]: `PostgreSQL`

File: config/postgresql/postgresql.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	Pw      string `envconfig:"POSTGRESQL_PW"`
+	Host    string `envconfig:"POSTGRESQL_HOST_NAME"`
+	Port    int    `envconfig:"POSTGRESQL_PORT"`
+	DBName  string `envconfig:"POSTGRESQL_DB_NAME"`
+	SSLMode string `envconfig:"POSTGRESQL_SSL_MODE"`
+}
+
+const (
+	// DefaultSSLMode is disabled
+	DefaultSSLMode = "disable"
```
Comment (jprobinson) [inline config/postgresql/postgresql.go]: For security sake, I think we'd want SSL on by default and required users to change the config to make it insecure.

**Comment by jprobinson** [comment]
Great start! Just a couple lil style things and the setting default `sslmode` to something higher. Let's default it to `verify-full`.

File: config/postgresql/postgresql.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+import (
+	"database/sql"
+	"fmt"
+	"net/url"
+
+	"github.com/NYTimes/gizmo/config"
+)
+
+// Config holds everything yo need to
```
Comment (jprobinson) [inline config/postgresql/postgresql.go]: s/yo/you

File: config/postgresql/postgresql.go [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+// connect and interact with a PostgreSQL DB
```
Comment (jprobinson) [inline config/postgresql/postgresql.go]: end with a period, please.

File: config/postgresql/postgresql.go [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+	User    string `envconfig:"POSTGRESQL_USER"`
+	Pw      string `envconfig:"POSTGRESQL_PW"`
+	Host    string `envconfig:"POSTGRESQL_HOST_NAME"`
+	Port    int    `envconfig:"POSTGRESQL_PORT"`
+	DBName  string `envconfig:"POSTGRESQL_DB_NAME"`
+	SSLMode string `envconfig:"POSTGRESQL_SSL_MODE"`
+}
+
+const (
+	// DefaultSSLMode is disabled
```
Comment (jprobinson) [inline config/postgresql/postgresql.go]: Change the comment to match the value

**Comment by jprobinson** [comment]
Looks good. Thanks, @theundeadmonk!

**Pull request merged**